Fixes for Gunicorn and Docker Build with Caches

sim4city opened the pull request:
Changes:

- Let the end user control how many Gunicorn worker processes run via the environment variable WEB_CONCURRENCY 

Modifies docker-compose.yml.template

- Now has a persistent Postgres database volume for reliability when the Postgres container was removed

Removes docker-compose.yml

- End user should rename docker-compose.yml.template to docker-compose.yml 
  - have their own specific version of docker-compose.yml, ignored by Git

Modifies .gitignore: append docker-compose.yml 

- Ignore this user specific file

Modifies the Dockerfile specifically for developers 

- Magnitude less time to rebuild the containers via the addition of experimental Docker BuildKit
  - Caches the data layers of project code, apt, and pip

  - First build runs everything and caches the layers.

  - Following builds only rebuild layers in which there are files with differences on disk, when compared to the files in the last build.
    - All other layers are copied from cache into the following build

Addresses issues: #33 and #21

Commit:
[FIX] #33 Update Gunicorn Params

Increased workers to 4 processes (about 8GB of memory in my case)

Commit:
[FIX] #21 Persistent Postgres Data

Postgres database data now in a persistent volume for reliability

Commit:
Improve Docker Build Speed

Added experimental Docker BuildKit layer caches

Cached project code, apt, and pip

Rebuild just the layers in which files were modified from the last build

Commit:
Merge docker-compose.yml.template

derneuere commented:
In #33 it states that the number of workers should be set to the number of (cpu threads * 2) + 1. 
Would it be possible to read out how many threads the system has and to adjust it based on that?
If that is not easily possible I would propose that we add a new environment variable to set in the docker-compose file.

Commit:
Update entrypoint.sh

By default, Gunicorn sets number of workers to the environment variable WEB_CONCURRENCY when it exists otherwise it starts one worker.

Commit:
Update docker-compose.yml.template

Added environment variable WEB_CONCURRENCY for end user to set the number of Gunicorn worker processes
